### PR TITLE
Registers package with sunpy template

### PR DESCRIPTION
**Pull request by CyclingNinja**

### CODE REVIEW

File: .black
Comment (Cadair) [inline .black:1]: nuke this file, let's just leave it in pyproject.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -48,48 +64,55 @@ jobs:
...
       pytest: false
-      default_python: '3.12'
       toxdeps: "tox-pypi-filter"
       libraries: |
         apt:
           - graphviz
       envs: |
         - linux: build_docs
 
-  online:
```
Comment (Cadair) [inline .github/workflows/ci.yml:66]: I suspect we need this online job and not the cron in this repo.

File: docs/conf.py [hunk 2]
```diff
@@ -1,24 +1,32 @@
...
+# This file does only contain a selection of the most common options. For a
+# full list see the documentation:
+# http://www.sphinx-doc.org/en/master/config
 
 import os
 import datetime
 from pathlib import Path
 
-from sunpy_sphinx_theme import PNG_ICON
+from sunpy_sphinx_theme import *
```
Comment (Cadair) [inline docs/conf.py:11]:
```suggestion
from sunpy_sphinx_theme import PNG_ICON
```

File: pyproject.toml [hunk 3]
```diff
@@ -1,11 +1,53 @@
 [build-system]
 requires = [
-  "setuptools",
-  "setuptools_scm[toml]",
+  "setuptools>=62.1",
+  "setuptools_scm[toml]>=6.2",
   "wheel",
   "oldest-supported-numpy",
```
Comment (Cadair) [inline pyproject.toml:6]: I don't think we need this here. @nabobalis there's no C extension in this repo right?

```suggestion
```

File: pyproject.toml [hunk 4]
```diff
@@ -1,11 +1,53 @@
...
+  "sphinx",
+  "sphinx-automodapi",
+  "sphinx-changelog",
+  "sphinx-copybutton",
+  "sphinx-gallery",
+  "sphinx-hoverxref",
+  "sphinxext-opengraph",
+  "sunpy-sphinx-theme",
+  # Remove next line when fixed in towncrier; see https://github.com/twisted/towncrier/issues/528
+  "importlib-resources",
```
Comment (Cadair) [inline pyproject.toml:46]: This is very fixed.
```suggestion
```

File: tox.ini [hunk 5]
```diff
@@ -1,32 +1,53 @@
...
 envlist =
-    py{310,311,312}{,-online,-sunpy,-oldestdeps}
-    build_docs
+    py{310,311,312}
+    py312-devdeps
+    py310-oldestdeps
     codestyle
+    build_docs
 requires =
     tox-pypi-filter
```
Comment (Cadair) [inline tox.ini:12]:
```suggestion
```

File: tox.ini [hunk 6]
```diff
@@ -43,22 +64,27 @@ deps =
...
+    -vvv \
+    -r fEs \
+    --pyargs drms \
+    --cov-report=xml \
+    --cov=drms \
+    --cov-config={toxinidir}/.coveragerc \
+    {toxinidir}/docs \
+    {posargs}
     !online: {env:PYTEST_COMMAND} {posargs}
     online: {env:PYTEST_COMMAND} --timeout=120 --remote-data=any {posargs}
```
Comment (Cadair) [inline tox.ini:86]:
```suggestion
    pytest \
    -vvv \
    -r fEs \
    --pyargs drms \
    --cov-report=xml \
    --cov=drms \
    --cov-config={toxinidir}/.coveragerc \
    online:  --timeout=120 \
    online:  --remote-data=any \
    {toxinidir}/docs \
    {posargs}
```

File: tox.ini [hunk 7]
```diff
@@ -43,22 +64,27 @@ deps =
...
     sunpy: zeep
 extras =
-    dev
-commands =
+    tests
+commands_pre =
+    oldestdeps: minimum_dependencies drms --filename requirements-min.txt
+    oldestdeps: pip install -r requirements-min.txt
     pip freeze --all --no-input
-    sunpy: pytest -vvv -s -ra --pyargs sunpy.net.jsoc --timeout=120 --remote-data=any {posargs}
```
Comment (Cadair) [inline tox.ini:49]: @nabobalis is there a good reason why this was in commands_pre? Do you want `-sunpy` to run both sunpy and drms tests?

File: tox.ini [hunk 8]
```diff
@@ -43,22 +64,27 @@ deps =
...
+commands =
+    # To amend the pytest command for different factors you can add a line
+    # which starts with a factor like `online: --remote-data=any \`
+    # If you have no factors which require different commands this is all you need:
+    pytest \
+    -vvv \
+    -r fEs \
+    --pyargs drms \
```
Comment (Cadair) [inline tox.ini:79]: Depending on how we want this to play out we can fold the sunpy job into this command like this maybe?
```suggestion
    !sunpy: --pyargs drms \
    sunpy: --pyargs sunpy.net.jsoc \
```

File: pyproject.toml [hunk 9]
```diff
@@ -1,11 +1,53 @@
...
+]
+dynamic = ["version"]
+
+[project.optional-dependencies]
+tests = [
+  "pytest",
+  "pytest-astropy",
+  "pytest-doctestplus",
+  "pytest-cov",
+  "pytest-xdist",
```
Comment (Cadair) [inline pyproject.toml:32]:
```suggestion
  "pytest-xdist",
  "astropy",
```

File: pytest.ini [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+doctest_plus = enabled
+doctest_optionflags =
+    NORMALIZE_WHITESPACE
+    FLOAT_CMP
+    ELLIPSIS
+text_file_format = rst
+addopts =
+    --doctest-rst
+    -p no:unraisableexception
+    -p no:threadexception
```
Comment (Cadair) [inline pytest.ini:25]:
```suggestion
    -p no:threadexception
markers =
    remote_data: marks this test function as needing remote data.
    jsoc: marks the test function as needing a connection to JSOC.
    kis: marks the test function as needing a connection to KIS.
    flaky: from sunpy
remote_data_strict = True
log_cli=true
log_level=INFO
```

File: pytest.ini [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+filterwarnings =
+    # Turn all warnings into errors so they do not pass silently.
+    error
+    # Do not fail on pytest config issues (i.e. missing plugins) but do show them
+    always::pytest.PytestConfigWarning
+    # A list of warnings to ignore follows. If you add to this list, you MUST
+    # add a comment or ideally a link to an issue that explains why the warning
+    # is being ignored
```
Comment (Cadair) [inline pytest.ini:33]:
```suggestion
    # is being ignored
    # This is due to dependencies building with a numpy version different from
    # the local installed numpy version, but should be fine
    # See https://github.com/numpy/numpy/issues/15748#issuecomment-598584838
    ignore:numpy.ufunc size changed:RuntimeWarning
    ignore:numpy.ndarray size changed:RuntimeWarning
    # https://github.com/pytest-dev/pytest-cov/issues/557
    ignore:The --rsyncdir command line argument and rsyncdirs config variable are deprecated.:DeprecationWarning
    ignore:.*is deprecated and slated for removal in Python 3:DeprecationWarning
```

File: .black [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+    | \.venv
+    | _build
+    | build
+    | dist
+    | docs
+    | .history
+  )/
+)
+'''
+
```
Comment (nabobalis) [inline .black:19]:
```suggestion
```

File: .black
Comment (nabobalis) [reply .black:1]: Nuke it and change the formatting to use ruff

File: pyproject.toml (hunk 3 above)
Comment (nabobalis) [reply pyproject.toml:6]: Yeah, no need. I did it for fun

File: tox.ini (hunk 7 above)
Comment (nabobalis) [reply tox.ini:49]: There is no good reason for this being here.

I was just lazy.

File: tox.ini (hunk 8 above)
Comment (nabobalis) [reply tox.ini:77]: Yeah that would be much better

File: .black
Comment (CyclingNinja) [reply .black:1]: Do you want the ruff to do Black things?

File: docs/conf.py (hunk 2 above)
Comment (CyclingNinja) [reply docs/conf.py:11]: in my experience so far, this doesn't work, but I may have been doing it wrong

Comment (nabobalis) [reply docs/conf.py:11]: Maybe you have an older version of the theme? That should work and if it doesn't, we need to fix it in the theme.

File: .black [hunk 13]
```diff
@@ -0,0 +1,19 @@
+line-length = 120
```
Comment (nabobalis) [inline .black:1]: We can delete this file.

File: .github/workflows/ci.yml [hunk 14]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (nabobalis) [inline .github/workflows/ci.yml:1]:
```suggestion
```
Does the gap come from the template

File: .github/workflows/ci.yml (hunk 1 above)
Comment (nabobalis) [reply .github/workflows/ci.yml:66]: yeah online stays as I need to test jsoc and the sunpy tests.

File: README.rst [hunk 15]
```diff
@@ -1,6 +1,5 @@
-====
-drms
-====
+Access HMI, AIA and MDI data from the Standford JSOC DRMS
+---------------------------------------------------------
```
Comment (nabobalis) [inline README.rst:2]:
```suggestion
====
drms
====

Access HMI, AIA and MDI data from the Stanford JSOC DRMS
```

File: docs/index.rst [hunk 16]
```diff
@@ -11,10 +11,18 @@ Python library for accessing HMI, AIA and MDI data from the Joint Science Operat
...
    generated/gallery/index
    citation
    whatsnew/index
+
+Indices and tables
+==================
+
+* :ref:`genindex`
+* :ref:`modindex`
+* :ref:`search`
```
Comment (nabobalis) [inline docs/index.rst:28]:
```suggestion
```
IS this in the template? We need to remove it.

**Comment by Cadair** [comment]
pre-commit.ci autofix

**Comment by Cadair** [comment]
@nabobalis The online build is failing with some errors in tutorial.rst and I am not sure if they are real errors or just weirdness.

**Comment by nabobalis** [comment]
That looks like a real error but I can't imagine it comes from this PR

File: docs/tutorial.rst [hunk 17]
```diff
@@ -224,8 +224,6 @@ The following, for example, only downloads the first file of the request:
 .. code-block:: python
 
     >>> export_request.download(out_dir, index=0)  # doctest: +REMOTE_DATA
-                                                  record                                                url                                           download
-    0  hmi.V_45s[2016.04.01_00:00:00_TAI][2]{Dopplerg...  http://jsoc.stanford.edu/SUM58/D803708321/S000...  ...
```
Comment (nabobalis) [inline docs/tutorial.rst:228]: Did this output nothing?

Comment (Cadair) [reply docs/tutorial.rst:228]: Well doctests don't capture log output apparently

Comment (nabobalis) [reply docs/tutorial.rst:228]: lmao

**Comment by Cadair** [comment]
:shipit: ?

**Comment by Cadair** [comment]
Well in that case, executive power deployed.

**Pull request merged**